### PR TITLE
Improve Responsive Layout and Sidebar Navigation

### LINKED ISSUE #78: Sidebar is not minimized if the website window get small.
**What happened**:
The sidebar on our website fails to minimize when the browser window is resized to smaller dimensions.

image

**What you expected to happen**:
Sidebar should auto minimized 
**How to reproduce it (as minimally and precisely as possible)**:

### PR DESCRIPTION
**Pull request by axif0**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #78 
**Special notes for your reviewer**:
Enhanced responsive layout and sidebar navigation with optimized route handling and performance improvements.

**Does this PR introduce a user-facing change?**:
Peek 2024-08-11 17-08
<!--
If no, just write "NONE" in the release-note block below.
If yes, a release note is required.

### CODE REVIEW

**Comment by warjiang** [comment]
@axif0 I think it's kind of feature that sidebar can collapse automatically according to the width of user, what do you think?

**Comment by axif0** [comment]
@warjiang ya.. you are right. Sorry i should have give `/kind feature` label.

File: ui/apps/dashboard/src/layout/index.tsx [hunk 1]
```diff
@@ -11,17 +11,39 @@ const { Sider: AntdSider, Content: AntdContent } = AntdLayout;
...
-      <AntdLayout className={cn('h-[calc(100vh-48px)]', 'overflow-hidden')}>
-        <AntdSider width={getSidebarWidth()}>
+      <AntdLayout className={cn('h-[calc(100vh-48px)]', 'overflow-hidden', 'flex')}>
+        <AntdSider
+          width={getSidebarWidth()}
+          collapsible
+          collapsed={collapsed}
+          breakpoint="lg"
+          onBreakpoint={(broken) => setCollapsed(broken)}
+          style={{ transition: 'width 0.3s ease' }}
```
Comment (warjiang) [inline ui/apps/dashboard/src/layout/index.tsx:41]: the antd lib already add the animation class, so it's no need to add duplicate style.
image

File: ui/apps/dashboard/src/layout/index.tsx [hunk 2]
```diff
@@ -11,17 +11,39 @@ const { Sider: AntdSider, Content: AntdContent } = AntdLayout;
...
+          trigger={null}
+        >
           <Sidebar />
         </AntdSider>
-        <AntdContent>
+        <AntdContent className="flex-grow" style={{ transition: 'margin-left 0.3s ease' }}>
```
Comment (warjiang) [inline ui/apps/dashboard/src/layout/index.tsx:46]: for the content, it seems that the animation doesn't work 🤩

File: ui/apps/dashboard/src/layout/index.tsx [hunk 3]
```diff
@@ -11,17 +11,39 @@ const { Sider: AntdSider, Content: AntdContent } = AntdLayout;
 
 export const MainLayout: FC = () => {
   const { authenticated } = useAuth();
+  const [collapsed, setCollapsed] = useState(false);
```
Comment (warjiang) [inline ui/apps/dashboard/src/layout/index.tsx:14]: here, it's implemented the similar logic twice, one in MainLayout, the other one in Sider, I think we can reduce one, only implement logic of checking collapse in the MainLayout, and pass the state of collapse the Sider component:
```
<Sider collapse={size?.width <= 768} />
```

**Comment by axif0** [comment]
Hello, @warjiang. Thank you for your detailed review. I learned a lot and have updated the PR according to your feedback. 😅

File: ui/apps/dashboard/src/layout/sidebar.tsx [hunk 4]
```diff
@@ -25,10 +30,15 @@ const Sidebar = () => {
       .map((m) => (m.handle as IRouteObjectHandle).sidebarKey);
   }, [matches]);
   return (
-    <div className={'w-full h-full overflow-y-auto'}>
+    <div
+      className={cn('w-full', 'h-full', 'overflow-y-auto', {
+        'collapsed-sidebar': collapsed,
+      })}
+      style={{ transition: 'width 0.3s' }}
```
Comment (warjiang) [inline ui/apps/dashboard/src/layout/sidebar.tsx:37]: when render to the dom, the ant Menu component already has `transition: width 0.3s cubic-bezier(0.2, 0, 0, 1) 0s;` style
image

File: ui/apps/dashboard/src/layout/sidebar.tsx [hunk 5]
```diff
@@ -25,10 +30,15 @@ const Sidebar = () => {
       .map((m) => (m.handle as IRouteObjectHandle).sidebarKey);
   }, [matches]);
   return (
-    <div className={'w-full h-full overflow-y-auto'}>
+    <div
+      className={cn('w-full', 'h-full', 'overflow-y-auto', {
+        'collapsed-sidebar': collapsed,
```
Comment (warjiang) [inline ui/apps/dashboard/src/layout/sidebar.tsx:35]: It seems that `collapsed-sidebar` class doesn't have any effect 😃 
Last time, i did not mention this, may it takes effect, can u explain it?

Comment (axif0) [reply ui/apps/dashboard/src/layout/sidebar.tsx:35]: Oh, the `Menu` component from `ant design` already handles the translation. Which is redundant. Thanks a lot.

**Pull request merged**